### PR TITLE
return nil when no node or service to avoid additional api call

### PR DESCRIPTION
**Pull request by allencloud**
**\- What I did**
1. return nil when no node to avoid API /info call
2. return nil when no service to avoid TaskList and NodeList api call
3. fix some typo in command confirmation

**\- How I did it**

**\- How to verify it**

**\- Description for the changelog**

**\- A picture of a cute animal (not mandatory but encouraged)**

Signed-off-by: allencloud allen.sun@daocloud.io

### CODE REVIEW

File: cli/command/node/list.go [hunk 1]
```diff
@@ -45,21 +45,29 @@ func newListCommand(dockerCli *command.DockerCli) *cobra.Command {
...
 		return err
 	}
 
-	info, err := client.Info(ctx)
+	if len(nodes) == 0 {
+		// if there is no node, no need to call /api endpoint and print directly.
+		goto SKIP
+	}
+
+	info, err = client.Info(ctx)
```
Comment (aaronlehmann) [inline cli/command/node/list.go]: Shouldn't `Info` also be skipped if `opts.quiet` is true? How about something like:

```go
if len(nodes) > 0 && !opts.quiet {
        info, err = client.Info(ctx)
        if err != nil {
                return err
        }
}
```

File: cli/command/service/list.go [hunk 2]
```diff
@@ -49,13 +49,21 @@ func newListCommand(dockerCli *command.DockerCli) *cobra.Command {
...
 	client := dockerCli.Client()
+	out := dockerCli.Out()
 
 	services, err := client.ServiceList(ctx, types.ServiceListOptions{Filter: opts.filter.Value()})
 	if err != nil {
 		return err
 	}
 
-	out := dockerCli.Out()
+	if len(services) == 0 {
```
Comment (aaronlehmann) [inline cli/command/service/list.go]: I'd move this into the `else` below.

File: cli/command/node/list.go (hunk 1 above)
Comment (allencloud) [inline cli/command/node/list.go]: Thanks for your review. @aaronlehmann 
I do not think goto is good, either. Now I updated this part, PTAL.

**Comment by aaronlehmann** [comment]
LGTM

**Comment by allencloud** [comment]
ping @vdemeester @thaJeztah 
Any update? 🐳

**Review by vdemeester — APPROVED** [review]
LGTM 👼

**Pull request merged**